mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/DeletedMessageConverter.java

```diff
@@ -109,10 +117,20 @@ private MaybeSender retrieveSender(Optional<Message> mimeMessage) {
             .map(AddressList::flatten)
             .flatMap(MailboxList::stream)
             .map(Mailbox::getAddress)
-            .map(Throwing.function(MailAddress::new))
+            .map(this::retrieveAddress)
+            .filter(mailAddress -> !mailAddress.asString().equals(NULL_SENDER_AS_STRING))
             .collect(Guavate.toImmutableList());
     }
 
+    private MailAddress retrieveAddress(String address) {
+        try {
+            return new MailAddress(address);
+        } catch (AddressException e) {
+            LOGGER.warn("Can not create the mailAddress from {}", address);
+            return MailAddress.nullSender();
```

Comment:
Nooooooooooooooooooooooooooooooo!!!

nullSender is a terrible idea to represent "not here".

Let's return an optional instead (or a STream that we can flatMap directly...